Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -878,7 +882,9 @@ extension NavigationViewController: NavigationServiceDelegate {
     }
     
     public func navigationService(_ service: NavigationService, shouldPreventReroutesWhenArrivingAt waypoint: Waypoint) -> Bool {
-        return navigationComponents.allSatisfy { $0.navigationService(service, shouldPreventReroutesWhenArrivingAt: waypoint) }
+        let componentsWantPreventReroutes = navigationComponents.allSatisfy { $0.navigationService(service, shouldPreventReroutesWhenArrivingAt: waypoint) }
+
+        return componentsWantPreventReroutes && (delegate?.navigationViewController(self, shouldPreventReroutesWhenArrivingAt: waypoint)) ?? RouteController.DefaultBehavior.shouldRerouteFromLocation
```

Comment:
This delegate method wasn't propagated to the `NavigationViewControllerDelegate`. Without returning `false` from here, reroutes after arrival won't work.

Reply:
As tail work, maybe we should replace DefaultBehavior with default method implementations on the respective protocols, in this case NavigationServiceDelegate and NavigationComponent. That way the default behavior happens automatically without us needing to remember to fall back to it.